### PR TITLE
bpo-27200: fix pathlib, ssl, turtle and weakref doctests

### PR DESCRIPTION
**Pull request by marco-buttu**
This PR is the last but one in the series [bpo-27200](https://bugs.python.org/issue27200).  The first one was #240, the second one #401, and the third #604.  To run the doctests (from the `Doc` directory):

```
$ sphinx-build -b doctest . build/doctest \
library/pathlib.rst \
library/ssl.rst \
library/turtle.rst \
library/weakref.rst
```

### CODE REVIEW

File: Doc/library/decimal.rst [hunk 1]
```diff
@@ -23,6 +23,11 @@
    # make sure each group gets a fresh context
    setcontext(Context())
 
+.. testcleanup:: *
+
+   # make sure other tests (also outside this file) get a fresh context
+   setcontext(Context())
+
```
Comment (marco-buttu) [inline Doc/library/decimal.rst]: The `setcontext(Context())` in `testsetup` does not guarantee that the tests outside this file get a fresh context.  We need to clean the context at the end of the tests.

Comment (berkerpeksag) [reply Doc/library/decimal.rst]: Please leave decimal.rst changes out of this PR and open a separate PR and then ping Stefan Krah.

File: Doc/library/shlex.rst [hunk 2]
```diff
@@ -43,15 +43,16 @@ The :mod:`shlex` module defines the following functions:
    string that can safely be used as one token in a shell command line, for
    cases where you cannot use a list.
 
-   This idiom would be unsafe::
+   This idiom would be unsafe:
```
Comment (berkerpeksag) [inline Doc/library/shlex.rst]: You have already modified this file in #604.

File: Doc/library/pathlib.rst [hunk 3]
```diff
@@ -46,7 +46,7 @@ Pure paths are useful in some special cases; for example:
 Basic use
 ---------
 
-Importing the main class::
+Importing the main class:
```
Comment (berkerpeksag) [inline Doc/library/pathlib.rst]: Please revert this change. The current version looks correct to me and the tests passed when I revert it to use ::.

**Review by berkerpeksag — CHANGES_REQUESTED** [review]
I think we also need to following change to ssl examples:

```diff
diff --git a/Doc/library/ssl.rst b/Doc/library/ssl.rst
index 01dbecd..02a0c79 100644
--- a/Doc/library/ssl.rst
+++ b/Doc/library/ssl.rst
@@ -2262,7 +2262,7 @@ SSL versions 2 and 3 are considered insecure and are therefore dangerous to
 use.  If you want maximum compatibility between clients and servers, it is
 recommended to use :const:`PROTOCOL_TLS_CLIENT` or
 :const:`PROTOCOL_TLS_SERVER` as the protocol version. SSLv2 and SSLv3 are
-disabled by default.
+disabled by default. ::
 
    >>> client_context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
    >>> client_context.options |= ssl.OP_NO_TLSv1
```

And there are the other failures:

```
**********************************************************************
File "library/ssl.rst", line 421, in newcontext
Failed example:
    timestamp
Expected:
    1515144883
Got:
    1515134083.0
**********************************************************************
File "library/ssl.rst", line 424, in newcontext
Failed example:
    print(datetime.utcfromtimestamp(timestamp))
Expected:
    2018-01-05 09:34:43
Got:
    2018-01-05 06:34:43
```

**Comment by marco-buttu** [comment]
Thank you very much for your time @berkerpeksag! I made all changes you requested me.  Just one question: you got a `float` from `ssl.cert_time_to_seconds()`.  Is it not a bug in the code?  Is it supposed to return both floats and ints?

**Comment by berkerpeksag** [comment]
I don't know, it's probably worth doing some VCS archaeology :) I built and ran tests in Python 3.4 with an ancient OpenSSL so this might be the culprit.

File: Doc/library/ssl.rst [hunk 4]
```diff
@@ -2257,7 +2262,7 @@ SSL versions 2 and 3 are considered insecure and are therefore dangerous to
 use.  If you want maximum compatibility between clients and servers, it is
 recommended to use :const:`PROTOCOL_TLS_CLIENT` or
 :const:`PROTOCOL_TLS_SERVER` as the protocol version. SSLv2 and SSLv3 are
-disabled by default.
+disabled by default. ::
```
Comment (berkerpeksag) [inline Doc/library/ssl.rst]: I know I suggested this style in my earlier comment, but could you move :: to a new line so we don't lose `git blame` information? This also applies to the changes in Doc/library/pathlib.rst.

Comment (marco-buttu) [reply Doc/library/ssl.rst]: Ok, done :-)  Thanks againt @berkerpeksag

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Thanks!

**Pull request merged**